### DISCUSSION TITLE
initramfs-linux.img much bigger than with mkinitcpio

### DISCUSSION THREAD

**Discussion by mabod** [dracutdevs/dracut#985] [General]
**Describe the bug**
The initramfs created by dracut is 3 times bigger than the img file I get from mkinitcpio. Is that expected / normal?

I am using zstd compression on both sides:

```
dracut:
-rw-r--r-- 1 root root  19M  6. Dez 08:38 initramfs-linux.img

mkinitcpio:
-rw-r--r-- 1 root root 6,1M  5. Dez 16:04 initramfs-linux.img-mkinitcpio
```
dracut commandline to create the file was:

`dracut --force --no-hostonly-cmdline --hostonly /boot/initramfs-linux.img --kver 5.9.12-arch1-1`

I dont know why this commandline shows "`--no-hostonly-cmdline --hostonly`" at the same time. This is coming automatically on arch. But anyways, even if I take out -`-no-hostonly-cmdline` or `--hostonly` the size is still 19M.

**Distribution used**
EndeavourOS

**Dracut version**
dracut 050-1

**Init system**
systemd 247-1

**Comment by barmadrid** [comment]
Arch's mkinitcpio default ZSTD compression setting is -19 compared to -15 of dracut. I believe that's the difference in size.

See [this](https://github.com/archlinux/mkinitcpio/blob/master/mkinitcpio#L215)
and [this](https://github.com/dracutdevs/dracut/blob/master/dracut.sh#L938)

**Comment by mabod** [comment]
That can not be it. Even with xz compression with mkinitcpio and with dracut the size difference is significant.

**Comment by johannbg** [comment]
To compare apples to apples regarding size and compression you will have to ensure that both resulting initramfs images contain the same content inside the initramfs image. 

Anyway converting the issue to discussion since a dialogue about performance, size, compression and general comparison belongs there ( as well as it prevents this issue being eaten up by the stale bot ). Once a consensus on how these might be improved ( or a saner default ) an enhancement issue and or better yet pr can be made to implement that consensus.
